File: api/tts.py
"""Deepgram TTS - Vercel Serverless Function"""
import json
import urllib.request

DG_KEY = 'ffbf23cb49159c3d5699587c20f1debecb819fc5'
DG_TTS_URL = 'https://api.deepgram.com/v1/speak'

def handler(request):
    if request.method == 'OPTIONS':
        return Response(
            status=204,
            body='',
            headers={
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Methods': 'POST, OPTIONS',
                'Access-Control-Allow-Headers': 'Content-Type'
            }
        )

    if request.method != 'POST':
        return Response(status=405, body='Method not allowed')

    try:
        body_data = json.loads(request.body)
        text = body_data.get('text', '')
        if not text:
            return Response(
                status=400,
                body=json.dumps({'error': 'No text provided'}),
                headers={'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*'}
            )

        payload = json.dumps({'text': text}).encode('utf-8')
        req = urllib.request.Request(
            DG_TTS_URL,
            data=payload,
            headers={
                'Authorization': 'Token ' + DG_KEY,
                'Content-Type': 'application/json'
            }
        )
        with urllib.request.urlopen(req, timeout=30) as resp:
            audio_data = resp.read()
            content_type = resp.headers.get('Content-Type', 'audio/mpeg')

        return Response(
            status=200,
            body=audio_data,
            headers={
                'Content-Type': content_type,
                'Access-Control-Allow-Origin': '*',
                'Content-Length': str(len(audio_data))
            }
        )
    except Exception as e:
        return Response(
            status=500,
            body=json.dumps({'error': str(e)}),
            headers={'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*'}
        )


class Response:
    def __init__(self, status=200, body='', headers=None):
        self.status = status
        self.body = body if isinstance(body, str) else body
        self.headers = headers or {'Content-Type': 'application/json'}
        if 'Content-Length' not in self.headers:
            b = self.body if isinstance(self.body, bytes) else self.body.encode('utf-8')
            self.headers['Content-Length'] = str(len(b))

File: api/tts.js
const https = require('https');

module.exports = async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') return res.status(204).end();
  if (req.method !== 'POST') return res.status(405).json({error:'Method not allowed'});

  const DG_KEY = process.env.DEEPGRAM_API_KEY || 'ffbf23cb49159c3d5699587c20f1debecb819fc5';

  try {
    const {text} = req.body;
    if (!text) return res.status(400).json({error:'No text'});

    const audio = await new Promise((resolve, reject) => {
      const r = https.request({
        hostname: 'api.deepgram.com',
        path: '/v1/speak',
        method: 'POST',
        headers: {
          'Authorization': 'Token ' + DG_KEY,
          'Content-Type': 'application/json'
        }
      }, resp => {
        const chunks = [];
        resp.on('data', c => chunks.push(c));
        resp.on('end', () => resolve(Buffer.concat(chunks)));
      });
      r.on('error', reject);
      r.write(JSON.stringify({text}));
      r.end();
    });

    res.setHeader('Content-Type', 'audio/mpeg');
    res.setHeader('Cache-Control', 'no-cache');
    res.send(audio);
  } catch(e) {
    res.status(500).json({error: e.message});
  }
};
